controller/feature.go

```diff
@@ -1,49 +1,65 @@
 package controller
 
 import (
+	jwtgo "github.com/dgrijalva/jwt-go"
 	"github.com/fabric8-services/fabric8-toggles-service/app"
-	"github.com/fabric8-services/fabric8-toggles-service/configuration"
 	"github.com/fabric8-services/fabric8-toggles-service/errorhandler"
+	"github.com/fabric8-services/fabric8-toggles-service/errors"
+	"github.com/fabric8-services/fabric8-toggles-service/featuretoggles"
+	"github.com/fabric8-services/fabric8-wit/log"
 	"github.com/goadesign/goa"
-	uuid "github.com/satori/go.uuid"
+	goajwt "github.com/goadesign/goa/middleware/security/jwt"
 )
 
 // FeatureController implements the feature resource.
 type FeatureController struct {
 	*goa.Controller
-	config *configuration.Data
+	client *featuretoggles.Client
 }
 
 // NewFeatureController creates a feature controller.
-func NewFeatureController(service *goa.Service, config *configuration.Data) *FeatureController {
+func NewFeatureController(service *goa.Service, client *featuretoggles.Client) *FeatureController {
 	return &FeatureController{
 		Controller: service.NewController("FeatureController"),
-		config:     config,
+		client:     client,
 	}
 }
 
 // Show runs the show action.
 func (c *FeatureController) Show(ctx *app.ShowFeatureContext) error {
 	// FeatureController_Show: start_implement
-	featureID, err := uuid.FromString(ctx.ID)
-	if err != nil {
-		return errorhandler.JSONErrorResponse(ctx, goa.ErrNotFound(err.Error()))
+	jwtToken := goajwt.ContextJWT(ctx)
+	if jwtToken == nil {
+		log.Error(ctx.Context, map[string]interface{}{}, "Unable to retrieve token")
+		return errorhandler.JSONErrorResponse(ctx, errors.NewUnauthorizedError("Missing JWT token"))
 	}
-	// TODO call unleash SDK to retrieve features/strategy
+	if groupID, ok := jwtToken.Claims.(jwtgo.MapClaims)["company"].(string); ok {
+		log.Info(ctx, nil, "Is feature id: %s enabled? ", ctx.ID)
+		enabledFeature := c.enabledFeature(ctx, ctx.ID, groupID)
```

Comment:
IMO, it would be simpler if you split the `enableFeature` function in 2 parts: 
- here, use the client to check if the feature is enabled (`enabled := c.client.IsFeatureEnabled(featureID, groupID)`)
- then call a utility function to convert the result into an `*app.Feature` 

this is what we usually do, so it's easier to follow the pattern here as well.